services/galley/src/Galley/API/LegalHold.hs

```diff
@@ -393,7 +393,11 @@ changeLegalholdStatus tid uid old new = do
     --
     UserLegalHoldDisabled -> case new of
       UserLegalHoldEnabled -> illegal
-      UserLegalHoldPending -> addblocks >> update
+      UserLegalHoldPending -> do
+        update
+        -- outcome of addblocks logic might depend on the update. we'd like to
+        -- fail the transition in case addblocks fails nevertheless
```

Comment:
What if `revert` fails?  Can we construct a hypothetical database state in memory and run `addblocks` on that instead?  Then we could maintain the old order.

Reply:
596293363c09e2c91c55809fe81086c12c33fccd

Reply:
Your solution doesn't rely on exceptions and state :+1: